### PR TITLE
Networking

### PR DESCRIPTION
**Pull request by StrongCoding**
merge networking and multiplexing part into main

### CODE REVIEW

File: Makefile
Comment (wrottger) [inline Makefile:1]: add -Wshadow back

File: browser.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <unistd.h> // For close()
+#include <sstream>
+#include <sys/epoll.h>  // für epoll_create1()
+#include <cerrno>
+
+// #define PORT 8080
```
Comment (wrottger) [inline browser.cpp:26]: remove #define

File: browser.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	// Send a message to the server
+	std::stringstream messageStream;
+	std::string message = "Hello, server!";
+	if (send(sockfd, message.c_str(), message.size(), 0) < 0)
+	{
+		std::cerr << "Failed to send the message" << errno << std::endl;
+		exit(EXIT_FAILURE);
+	}
+	// if (send(sockfd, NULL,0, 0) < 0)
```
Comment (wrottger) [inline browser.cpp:67]: remove

**Review by stevebalk — APPROVED** [review]
Way to much stuff

File: incl/SocketHandling.hpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# include <stdexcept>
+# include "Config.hpp"
+
+class SocketHandling
+{
+	private:
+		std::vector<configObject> &_config;
+		std::vector<int> _openFds;
+		int	_epollFd;
+
```
Comment (wrottger) [inline incl/SocketHandling.hpp:25]: remove '_' before name

File: src/EventHandler.cpp [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+					std::string responseBody = "<!DOCTYPE html><html><head><title>Hello World</title></head>"
+												"<body><h1>Hello, World!</h1></body></html>";
+
+					std::ostringstream oss;
+					oss << responseBody.size();
+
+					std::string httpResponse = "HTTP/1.1 200 OK\r\n"
+												"Content-Type: text/html; charset=UTF-8\r\n"
+												"Content-Length: " +
+							oss.str() + "\r\n\r\n" + responseBody;
```
Comment (wrottger) [inline src/EventHandler.cpp:56]: A bit deep intendation. Maybe its own function?

**Review by wrottger — COMMENTED** [review]
Very nice. Some small visual improvements possible.

**Pull request merged**